Sources/Private/Experimental/Layers/ImageLayer.swift

```diff
@@ -0,0 +1,50 @@
+// Created by Cal Stephens on 1/10/22.
+// Copyright © 2022 Airbnb Inc. All rights reserved.
+
+import QuartzCore
+
+/// The `CALayer` type responsible for rendering `ImageLayerModel`s
+final class ImageLayer: BaseCompositionLayer {
+
+  // MARK: Lifecycle
+
+  init(
+    imageLayer: ImageLayerModel,
+    context: LayerContext)
+  {
+    self.imageLayer = imageLayer
+    super.init(layerModel: imageLayer)
+    setupSublayers(context: context)
+  }
+
+  required init?(coder _: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  /// Called by CoreAnimation to create a shadow copy of this layer
+  /// More details: https://developer.apple.com/documentation/quartzcore/calayer/1410842-init
+  override init(layer: Any) {
+    guard let layer = layer as? Self else {
+      fatalError("\(Self.self).init(layer:) incorrectly called with \(type(of: layer))")
+    }
+
+    imageLayer = layer.imageLayer
+    super.init(layer: layer)
+  }
+
+  // MARK: Private
+
+  private let imageLayer: ImageLayerModel
+
+  private func setupSublayers(context: LayerContext) {
+    guard
+      let imageAsset = context.assetLibrary?.imageAssets[imageLayer.referenceID],
+      let image = context.imageProvider.imageForAsset(asset: imageAsset)
+    else { return }
```

Comment:
Good question, some thoughts:
 1. The legacy rendering engine silently ignores cases where referenced objects don't exist (e.g. parent layer by index, pre-comp layer assets, image assets) so it feels somewhat ok to continue doing that in the new engine.
 2. I think there's a somewhat reasonable expectation that the json will be well-formed, since they're typically output directly by bodymovin. 
Open to changing this if you think it's a good idea (if so we could also update parent layer and pre-comp layer asset lookup to log on failure)

Reply:
I definitely agree that silent failures make sense in the case where the JSON is malformed and the error would not be recoverable by the client developer without exporting new JSON. However in cases where we have client engineer programmer error (e.g. the asset wasn't provided by the image provider) I think it would make sense to log as it would enable a client engineer to more easily diagnose why an animation isn't rendering correctly. I could also imagine a system like the [`EpoxyLogger`](https://git.musta.ch/airbnb/epoxy-ios/blob/master/Sources/EpoxyCore/Logging/EpoxyLogger.swift) to support logging warnings in a way that consumers can pipe into their own logging/alerting systems for assertions.

Reply:
Adding a logging system like `EpoxyLogger` seems like a good idea